### PR TITLE
fix: prevent ignorance of initial changes

### PR DESCRIPTION
**Pull request by dos65**

### CODE REVIEW

File: backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala [hunk 1]
```diff
@@ -84,7 +84,10 @@ private final class BloopNameHashing(
           initialChangedSources,
           allSources.map(v => v: VirtualFileRef),
           previous
-        ).collect { case f: VirtualFile => f }
```
Comment (dos65) [inline backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala:87]: `initialChagedSources` are received as `VirtualFileRef` and `mapInvalidationToSources` doesn't lift them into `VirtualFile`.
Using `collect` here is incorrect as it just drops correctly invalidated sources.

File: backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala [hunk 2]
```diff
@@ -84,7 +84,10 @@ private final class BloopNameHashing(
...
+        ).map {
+          case f: VirtualFile => f
+          case ref: VirtualFileRef => PlainVirtualFileConverter.converter.toVirtualFile(ref)
+        }
```
Comment (dos65) [inline backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala:90]: @tgodzik Is `PlainVirtualFileConverter.converter` usage fine here?
I've seen that in some methods `converter` is passed as parameter.

Comment (tgodzik) [reply backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala:90]: I think so, we don't use any other converters currently, so this should be fine.

File: backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala (hunk 1 above)
Comment (tgodzik) [reply backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala:87]: och! That looks like the issue i've been having! Awesom!

**Review by tgodzik — APPROVED** [review]
LGTM!

File: backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala (hunk 2 above)
Comment (tgodzik) [reply backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala:90]: what is the actual type returned? VirtualFileRef is only an interface, no?

**Pull request merged**